selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java

```diff
@@ -212,14 +215,13 @@ private String getWsAgentUrl(String workspaceId) throws Exception {
     Workspace workspace = workspaceServiceClient.getById(workspaceId);
     workspaceServiceClient.ensureRunningStatus(workspace);
 
-    return workspace
-            .getRuntime()
-            .getMachines()
-            .get(0)
-            .getRuntime()
-            .getServers()
-            .get(String.valueOf(WS_AGENT_PORT) + "/tcp")
-            .getUrl()
-        + "/project";
+    Map<String, ? extends Machine> machines =
+        workspaceServiceClient.getById(workspaceId).getRuntime().getMachines();
+    for (Machine machine : machines.values()) {
+      if (containsWsAgentServer(machine)) {
+        return machine.getServers().get(SERVER_WS_AGENT_HTTP_REFERENCE).getUrl() + "/project";
```

Comment:
It is not likely, but still possible to get NPE here in case machine doesn't have `SERVER_WS_AGENT_HTTP_REFERENCE`.
Can you check for nullness for this case before retrieving the URL?

Reply:
ok